### ISSUE TITLE
Model stream deletion events as a `Delete` activity

### ISSUE THREAD

**Issue by p-kimberley** [gchq/stroom#4146]
Version: 7.3

A stream deletion event is currently output like:
```xml
<EventDetail>
  <TypeId>MetaResourceImpl.updateStatus</TypeId>
  <Description>Modify the status of selected streams to Deleted</Description>
  <Search>
    <Query>
      <Advanced>
        <And>
          <Term>
            <Name>Id</Name>
            <Condition>Equals</Condition>
            <Value>2991067</Value>
          </Term>
        </And>
      </Advanced>
    </Query>
  </Search>
</EventDetail>
```

This is missing two key pieces of information:
* Feed name
* Stream type

Additionally, a deletion event should be modelled as a `Delete` activity, for ease of discovery/search.

**Comment by at055612** [comment]
Results of changes are as follows:

Delete of single stream.
```xml
<EventDetail>
  <TypeId>MetaResourceImpl.updateStatus</TypeId>
  <Description>Delete stream with ID 920</Description>
  <Delete>
    <Object>
      <Id>920</Id>
      <State>Unlocked</State>
    </Object>
    <Data Name="Feed" Value="DATA_SPLITTER-EVENTS"/>
    <Data Name="Type" Value="Error"/>
  </Delete>
</EventDetail>
```

Restore single stream
```xml
<EventDetail>
  <TypeId>MetaResourceImpl.updateStatus</TypeId>
  <Description>Update the status of stream with ID 1999 from Deleted to Unlocked</Description>
  <Update>
    <Before>
      <Object>
        <Id>1999</Id>
        <State>Deleted</State>
      </Object>
    </Before>
    <After>
      <Object>
        <Id>1999</Id>
        <State>Unlocked</State>
      </Object>
    </After>
    <Data Name="Feed" Value="DATA_SPLITTER-EVENTS"/>
    <Data Name="Type" Value="Error"/>
  </Update>
</EventDetail>
```

Delete N checked streams
```xml
<EventDetail>
  <TypeId>MetaResourceImpl.updateStatus</TypeId>
  <Description>Delete 3 streams</Description>
  <Delete>
    <Criteria>
      <Query>
        <Simple>
          <Include>1985,1976,1975</Include>
        </Simple>
      </Query>
    </Criteria>
    <Data Name="Count" Value="3"/>
    <Data Name="FeedCount" Value="1"/>
    <Data Name="TypeCount" Value="2"/>
    <Data Name="ProcessorCount" Value="1"/>
    <Data Name="PipelineCount" Value="1"/>
    <Data Name="StatusCount" Value="1"/>
    <Data Name="MinCreateTime" Value="2024-02-08T15:17:25.788Z"/>
    <Data Name="MaxCreateTime" Value="2024-02-08T15:17:26.605Z"/>
  </Delete>
</EventDetail>
```

Delete using selectAll & filter
```xml
<EventDetail>
  <TypeId>MetaResourceImpl.updateStatus</TypeId>
  <Description>Delete streams matching a criteria</Description>
  <Delete>
    <Criteria>
      <Query>
        <Advanced>
          <And>
            <Term>
              <Name>Status</Name>
              <Condition>Equals</Condition>
              <Value>Unlocked</Value>
            </Term>
            <Term>
              <Name>Type</Name>
              <Condition>Equals</Condition>
              <Value>Error</Value>
            </Term>
            <Term>
              <Name>Create Time</Name>
              <Condition>GreaterThan</Condition>
              <Value>2024-02-22T00:00:00.000Z</Value>
            </Term>
          </And>
        </Advanced>
      </Query>
    </Criteria>
    <Data Name="Count" Value="5"/>
    <Data Name="FeedCount" Value="1"/>
    <Data Name="TypeCount" Value="1"/>
    <Data Name="ProcessorCount" Value="1"/>
    <Data Name="PipelineCount" Value="1"/>
    <Data Name="StatusCount" Value="1"/>
    <Data Name="MinCreateTime" Value="2024-02-29T14:06:21.338Z"/>
    <Data Name="MaxCreateTime" Value="2024-03-01T12:10:26.397Z"/>
  </Delete>
</EventDetail>
```

Restore using selectAll & filter
```xml
<EventDetail>
  <TypeId>MetaResourceImpl.updateStatus</TypeId>
  <Description>Update the status of streams matching a criteria from Deleted to Unlocked</Description>
  <Update>
    <After>
      <Criteria>
        <Query>
          <Advanced>
            <And>
              <Term>
                <Name>Status</Name>
                <Condition>Equals</Condition>
                <Value>Deleted</Value>
              </Term>
              <Term>
                <Name>Create Time</Name>
                <Condition>GreaterThan</Condition>
                <Value>2024-02-22T00:00:00.000Z</Value>
              </Term>
            </And>
          </Advanced>
        </Query>
      </Criteria>
    </After>
    <Data Name="Count" Value="6"/>
    <Data Name="FeedCount" Value="2"/>
    <Data Name="TypeCount" Value="2"/>
    <Data Name="ProcessorCount" Value="2"/>
    <Data Name="PipelineCount" Value="2"/>
    <Data Name="StatusCount" Value="1"/>
    <Data Name="MinCreateTime" Value="2024-02-28T09:34:36.040Z"/>
    <Data Name="MaxCreateTime" Value="2024-03-01T12:10:26.397Z"/>
  </Update>
</EventDetail>
```

**Comment by at055612** [comment]
Waiting on @p-kimberley to confirm he is happy with the event structure generated.

**Comment by at055612** [comment]
Changed the data items to include the distinct feeds/types/statuses.



```xml
<EventDetail>
  <TypeId>MetaResourceImpl.updateStatus</TypeId>
  <Description>Delete 2 streams</Description>
  <Delete>
    <Criteria>
      <Query>
        <Simple>
          <Include>2029,1923</Include>
        </Simple>
      </Query>
    </Criteria>
    <Data Name="Count" Value="2"/>
    <Data Name="FeedCount" Value="2"/>
    <Data Name="TypeCount" Value="1"/>
    <Data Name="ProcessorCount" Value="2"/>
    <Data Name="PipelineCount" Value="2"/>
    <Data Name="StatusCount" Value="1"/>
    <Data Name="Feeds">
      <Data Name="Feed" Value="TEST_REFERENCE_DATA-EVENTS"/>
      <Data Name="Feed" Value="ZIP_TEST-DATA_SPLITTER-EVENTS"/>
    </Data>
    <Data Name="Types">
      <Data Name="Type" Value="Error"/>
    </Data>
    <Data Name="Statuses">
      <Data Name="Status" Value="Unlocked"/>
    </Data>
    <Data Name="MinCreateTime" Value="2023-12-29T11:09:09.485Z"/>
    <Data Name="MaxCreateTime" Value="2024-02-29T17:48:51.692Z"/>
  </Delete>
</EventDetail>
```

Only shows up to 20 distinct values.  If truncated you will see 

```xml
    <Data Name="FeedCount" Value="55"/>
    <!-- ... -->
    <Data Name="Feeds">
      <Data Name="Feed" Value="TEST_REFERENCE_DATA-EVENTS"/>
      <!-- ...18 data items... -->
      <Data Name="Feed" Value="ZIP_TEST-DATA_SPLITTER-EVENTS"/>
      <Data Name="IsListTruncated" Value="true"/>
    </Data>
```

I have also changed the selection summary popup to show the distinct types

![image](https://github.com/gchq/stroom/assets/22818309/426d8293-aa67-4245-bd7d-2dd04dc74742)

**Comment by at055612** [comment]
Fixed in >7.5-beta.14
